nautobot/core/api/serializers.py

```diff
@@ -122,13 +123,50 @@ class BaseModelSerializer(OptInFieldsMixin, serializers.HyperlinkedModelSerializ
     display = serializers.SerializerMethodField(read_only=True, help_text="Human friendly display value")
     object_type = ObjectTypeField()
     composite_key = serializers.SerializerMethodField()
+    natural_keys_values = []
+    related_model_fields_names = []
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # If it is not a Nested Serializer, we should set the depth argument to whatever is in the request's context
         if not self.is_nested:
             self.Meta.depth = self.context.get("depth", 0)
 
+        # Check if the request is related to CSV export;
+        if self._is_csv_request() and self.instance:
+            related_fields_natural_key_field_lookups = self._get_related_fields_natural_key_field_lookups()
+            if isinstance(self.instance, QuerySet):
+                queryset = self.instance
+            else:
+                # We would only need to run one additional query, making this a more efficient method of
+                # obtaining all the natural key values for this instance;
+                queryset = self.instance.__class__.objects.filter(pk=self.instance.pk)
+            self.natural_keys_values = queryset.values(*related_fields_natural_key_field_lookups, "pk")
+
+    def _get_related_fields_natural_key_field_lookups(self):
+        """Retrieve a list of field lookups for natural key fields of related models.
+
+        This method iterates through the related fields of the Serializer model,
+        retrieves the natural_key_field_lookups for each related model, and prepends the field name
+        to create a list of field lookups.
+        """
+        model = self.Meta.model
+        field_lookups = []
+        fields = [field for field in model._meta.get_fields() if field.is_relation]
+        self.related_model_fields_names = [field.name for field in fields]
+        for field in fields:
+            # Get each related field model's natural_key_fields and prepend field name
+            with contextlib.suppress(Exception):
+                field_lookups.extend(
+                    f"{field.name}__{lookup}" for lookup in field.related_model.natural_key_field_lookups
+                )
```

Comment:
Perhaps related to the above comment, do we need to have special-case handling for relations to non-Nautobot models (`Group`, `ContentType`) that don't have a `natural_key_field_lookups` attribute?